cpp/src/c_api/graph_mg.cpp

```diff
@@ -211,7 +211,7 @@ extern "C" cugraph_error_code_t cugraph_mg_graph_create(
                "Invalid input arguments: src type != dst type.",
                *error);
 
-  CAPI_EXPECTS(!weights || (p_weights->size_ == p_src->size_),
+  CAPI_EXPECTS((weights == nullptr) || (p_weights->size_ == p_src->size_),
                CUGRAPH_INVALID_INPUT,
                "Invalid input arguments: src size != weights size.",
                *error);
```

Comment:
Question, `weights` can be `nullptr` for either unweighted graphs or weighted but # (local) edges == 0. How do you distinguish these two cases?

Reply:
The assumption in the C API is the following:
 * For an unweighted graph, `weights == nullptr`
 * For a weighted graph, the weights will be passed, but that it will be an empty array if the number of local edges is 0.  So the `cugraph_type-erased_device_array_view_t *` would be passed if the graph is weighted, but the data pointer will be `nullptr` and the size will be 0

I will add that to the documentation of the graph creation functions.